security-analysis/src/main/java/com/powsybl/security/Security.java

```diff
@@ -351,16 +575,186 @@ private static LimitViolationKey toKey(LimitViolation violation) {
+                        .filter(violation -> preContingencyViolations.isEmpty() || !preContingencyViolations.contains(toKey(violation)))
+                        .collect(Collectors.toList());
+
+                if (!filteredLimitViolations2.isEmpty() || !postContingencyResult.getLimitViolationsResult().isComputationOk()) {
+                    formatter.writeCell(postContingencyResult.getContingency().getId())
+                            .writeCell(postContingencyResult.getLimitViolationsResult().isComputationOk() ? "converge" : "diverge")
+                            .writeEmptyCell()
+                            .writeEmptyCell()
+                            .writeEmptyCell()
+                            .writeCell("(" + filteredLimitViolations2.size() + ")")
+                            .writeEmptyCell()
+                            .writeEmptyCell()
+                            .writeEmptyCell()
+                            .writeEmptyCell()
+                            .writeEmptyCell()
+                            .writeEmptyCell()
+                            .writeEmptyCell();
+
+
+                    for (String action : postContingencyResult.getLimitViolationsResult().getActionsTaken()) {
+                        formatter.writeEmptyCell()
+                                .writeEmptyCell()
+                                .writeCell(action)
+                                .writeEmptyCell()
+                                .writeEmptyCell()
+                                .writeEmptyCell()
+                                .writeEmptyCell()
+                                .writeEmptyCell()
+                                .writeEmptyCell()
+                                .writeEmptyCell()
+                                .writeEmptyCell()
+                                .writeEmptyCell()
+                                .writeEmptyCell();
+                    }
+
+                    filteredLimitViolations2.stream()
+                            .sorted(Comparator.comparing(LimitViolation::getSubjectId))
+                            .forEach(writeLimitViolation(network, formatter));
+                }
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        };
+    }
+
+    private static Consumer<? super LimitViolation> writeLimitViolation(Network network, TableFormatter formatter) {
+        return violation -> {
+            try {
+                formatter.writeEmptyCell()
+                        .writeEmptyCell()
+                        .writeEmptyCell()
+                        .writeCell(LimitViolation.getCountry(violation, network) != null ? LimitViolation.getCountry(violation, network).name() : "")
+                        .writeCell(Float.isNaN(LimitViolation.getNominalVoltage(violation, network)) ? "" : Integer.toString((int) LimitViolation.getNominalVoltage(violation, network)))
```

Comment:
LimitViolation.getNominalVoltage cannot be NaN

Reply:
I've removed the test Float.isNaN on the LimitViolation.getNominalVoltage. Done.